fields/types/categorySet/CategorySetType.js

```diff
@@ -0,0 +1,49 @@
+var FieldType = require('../Type');
+var TextType = require('../text/TextType');
+var keystone = require('../../../');
+var util = require('util');
+
+/**
+ * HTML FieldType Constructor
+ * @extends Field
+ * @api public
+ */
+function categorySet(list, path, options) {
+  this._nativeType = [{
+    category: keystone.mongoose.Schema.Types.ObjectId,
+    subcategory: keystone.mongoose.Schema.Types.ObjectId
+  }];
+  this._defaultSize = 'full';
+  categorySet.super_.call(this, list, path, options);
+}
+util.inherits(categorySet, FieldType);
+
+/* Inherit from TextType prototype */
+categorySet.prototype.addFilterToQuery = TextType.prototype.addFilterToQuery;
+
+/**
+ * Updates the value for this field in the item from a data object
+ *
+ * Will accept either the field path, or paths.draft
+ *
+ * @api public
+ */
+
+categorySet.prototype.updateItem = function(item, data, callback) {
+  var value = this.getValueFromData(data);
+  if (value && value !== '') {
+    var dObj = {};
+    try {
+      dObj = JSON.parse(value);
```

Comment:
Could we add some check with category set item in dObj to filter out `{ category: undefined, subcategory: undefined }` case ?

Reply:
I filter that ahead in the `<input>` component via `JSON.stringify(value.filter(categorySet => categorySet && categorySet.category))`
https://github.com/twreporter/keystone/blob/4f6a9726546c3f5517856ec16f5370280950f2d8/fields/types/categorySet/CategorySet.js#L164

but fine to reinforce again in https://github.com/twreporter/keystone/pull/220/commits/b42d8ecaa70568b48f29e4add9a60bc87441db31